### PR TITLE
Issue #1198 by JrgenGNielsen : no user login redirect if in maintaina…

### PR DESCRIPTION
**Pull request by JrgenGNielsen**
…nce mode

### CODE REVIEW

File: ding_user.module [hunk 1]
```diff
@@ -24,7 +24,10 @@ function ding_user_preprocess_html(&$variables) {
     elseif (count($args) == 1 || isset($args[1]) && in_array($args[1], array('login', 'password'))) {
       // Redirect anonymous user to front page login. This is to prevent the
       // library user from seeing /user, login and password page.
-      drupal_goto('', array('fragment' => 'login'));
+      // Unless in maintenance_mode
+      if ( !variable_get('maintenance_mode', 0) ) {
```
Comment (kasperg) [inline ding_user.module]: A few whitespace issues here.

File: ding_user.module [hunk 2]
```diff
@@ -24,7 +24,10 @@ function ding_user_preprocess_html(&$variables) {
     elseif (count($args) == 1 || isset($args[1]) && in_array($args[1], array('login', 'password'))) {
       // Redirect anonymous user to front page login. This is to prevent the
       // library user from seeing /user, login and password page.
-      drupal_goto('', array('fragment' => 'login'));
+      // Unless in maintenance_mode
```
Comment (kasperg) [inline ding_user.module]: Please add a inline comment explaining *why* we do not want this redirect in maintenance mode.

**Comment by kasperg** [comment]
@JrgenGNielsen: Review complete. I have a couple of minor quibbles.

File: ding_user.module [hunk 3]
```diff
@@ -24,7 +24,10 @@ function ding_user_preprocess_html(&$variables) {
     elseif (count($args) == 1 || isset($args[1]) && in_array($args[1], array('login', 'password'))) {
       // Redirect anonymous user to front page login. This is to prevent the
       // library user from seeing /user, login and password page.
-      drupal_goto('', array('fragment' => 'login'));
+      // Unless in maintenance_mode: There's no front page login when in maintenance_mode, and a redirect prevent login.
+      if (!variable_get('maintenance_mode', 0)) {
+        drupal_goto('', array('fragment' => 'login'));
+      }
```
Comment (JrgenGNielsen) [inline ding_user.module]: @kasperg : Updated comment, and removed whitespace

**Comment by kasperg** [comment]
Reviewed and approved. Thanks!

**Pull request merged**